packages/flutter_tools/bin/xcode_backend.dart

```diff
@@ -0,0 +1,466 @@
+      runSync(
+        'rsync',
+        <String>[
+          '-av',
+          '--delete',
+          '--filter',
+          '- .DS_Store',
+          flutterFramework,
+          '$derivedDir/engine',
+        ],
+        verbose: verbose,
+      );
+    }
+
+    final List<String> flutterArgs = <String>[];
+
+    if (verbose) {
+      flutterArgs.add('--verbose');
+    }
+
+    if (environment['FLUTTER_ENGINE'] != null && environment['FLUTTER_ENGINE']!.isNotEmpty) {
+      flutterArgs.add('--local-engine-src-path=${environment['FLUTTER_ENGINE']}');
+    }
+
+    if (environment['LOCAL_ENGINE'] != null && environment['LOCAL_ENGINE']!.isNotEmpty) {
+      flutterArgs.add('--local-engine=${environment['LOCAL_ENGINE']}');
+    }
+
+    flutterArgs.addAll(<String>[
+      'assemble',
+      '--no-version-check',
+      '--output=${environment['BUILT_PRODUCTS_DIR'] ?? ''}/',
+      '-dTargetPlatform=ios',
+      '-dTargetFile=$targetPath',
+      '-dBuildMode=$buildMode',
+      '-dIosArchs=${environment['ARCHS'] ?? ''}',
+      '-dSdkRoot=${environment['SDKROOT'] ?? ''}',
+      '-dSplitDebugInfo=${environment['SPLIT_DEBUG_INFO'] ?? ''}',
+      '-dTreeShakeIcons=${environment['TREE_SHAKE_ICONS'] ?? ''}',
+      '-dTrackWidgetCreation=${environment['TRACK_WIDGET_CREATION'] ?? ''}',
+      '-dDartObfuscation=${environment['DART_OBFUSCATION'] ?? ''}',
+      '-dEnableBitcode=$bitcodeFlag',
+      '--ExtraGenSnapshotOptions=${environment['EXTRA_GEN_SNAPSHOT_OPTIONS'] ?? ''}',
+      '--DartDefines=${environment['DART_DEFINES'] ?? ''}',
+      '--ExtraFrontEndOptions=${environment['EXTRA_FRONT_END_OPTIONS'] ?? ''}',
+    ]);
+
+    if (environment['PERFORMANCE_MEASUREMENT_FILE'] != null && environment['PERFORMANCE_MEASUREMENT_FILE']!.isNotEmpty) {
+      flutterArgs.add('--performance-measurement-file=${environment['PERFORMANCE_MEASUREMENT_FILE']}');
+    }
+
+    final String? expandedCodeSignIdentity = environment['EXPANDED_CODE_SIGN_IDENTITY'];
+    if (expandedCodeSignIdentity != null && expandedCodeSignIdentity.isNotEmpty && environment['CODE_SIGNING_REQUIRED'] != 'NO') {
+      flutterArgs.add('-dCodesignIdentity=$expandedCodeSignIdentity');
+    }
+
+    if (environment['BUNDLE_SKSL_PATH'] != null && environment['BUNDLE_SKSL_PATH']!.isNotEmpty) {
+      flutterArgs.add('-dBundleSkSLPath=${environment['BUNDLE_SKSL_PATH']}');
+    }
+
+    if (environment['CODE_SIZE_DIRECTORY'] != null && environment['CODE_SIZE_DIRECTORY']!.isNotEmpty) {
+      flutterArgs.add('-dCodeSizeDirectory=${environment['CODE_SIZE_DIRECTORY']}');
+    }
+
+    flutterArgs.add('${buildMode}_ios_bundle_flutter_assets');
+
+    final ProcessResult result = runSync(
+      '${environmentEnsure('FLUTTER_ROOT')}/bin/flutter',
+      flutterArgs,
+      verbose: verbose,
+      allowFail: true,
+      workingDirectory: projectPath, // equivalent of RunCommand pushd "${project_path}"
+    );
+
+    if (result.exitCode != 0) {
+      echo(result.stdout as String);
```

Comment:
As far as I can tell, the only behavioral difference between after this change are these additional echoes before exiting if `flutter assemble` failed.

Does this satisfy:

> Would be really awesome also to get the logging piped from assemble back into the Xcode logs--right now it just gets eaten and you get an exit code in Xcode and you don't know why it failed.

https://github.com/flutter/flutter/issues/84382#issuecomment-859926242

Reply:
It's outputting a bunch of newlines when assemble fails, which is unexpected:
<img width="388" alt="Screen Shot 2021-07-23 at 3 00 30 PM" src="https://user-images.githubusercontent.com/682784/126845608-8c8283c7-09d7-4766-b534-0f8f4135379b.png">

To reproduce I changed 
https://github.com/flutter/flutter/blob/0de6bd41ba880195725144d11e5c9e378a497ca7/packages/flutter_tools/lib/src/build_system/targets/ios.dart#L224-L226
to instead throw:
```dart
    if (environment.defines[kSdkRoot] != null) {
      throw MissingDefineException(kSdkRoot, name);
    }
```

> Would be really awesome also to get the logging piped from assemble back into the Xcode logs

I would hope to see "Target debug_universal_framework required define SdkRoot but it was not provided" in the Xcode logs or `flutter build ios` output.  But I think feature parity is good enough for this PR.

Reply:
I removed the extraneous newlines. Not seeing assemble failures (except in verbose mode) will have to be fixed in the assemble command.